scripts/ioc_deploy.py

```diff
@@ -233,30 +426,24 @@ def finalize_name(name: str, github_org: str, ioc_dir: str, verbose: bool) -> st
     logger.info(f"{repo_dir} does not exist, checking for other casings")
     _, area, suffix = split_ioc_name(name)
     # First, check for casing on area
-    found_area = False
-    for path in Path(ioc_dir).iterdir():
-        if path.name.lower() == area.lower():
-            area = path.name
-            found_area = True
-            logger.info(f"Using {area} as the area")
-            break
-    if not found_area:
+    try:
+        area = find_casing_in_dir(dir=ioc_dir, name=area)
+    except RuntimeError:
         logger.info("This is a new area, checking readme for casing")
         name = casing_from_readme(name=name, readme_text=readme_text)
         logger.info(f"Using casing: {name}")
         return name
+    logger.info(f"Using {area} as the area")
 
-    found_suffix = False
-    for path in (Path(ioc_dir) / area).iterdir():
-        if path.name.lower() == suffix.lower():
-            suffix = path.name
-            found_suffix = True
-            logger.info(f"Using {suffix} as the name")
-            break
-    if not found_suffix:
+    try:
+        suffix = find_casing_in_dir(dir=str(Path(ioc_dir) / area), name=suffix)
+    except RuntimeError:
         logger.info("This is a new ioc, checking readme for casing")
         # Use suffix from readme but keep area from directory search
-        suffix = split_ioc_name(casing_from_readme(name=name, readme_text=readme_text))[2]
+        suffix = split_ioc_name(casing_from_readme(name=name, readme_text=readme_text))[
+            2
+        ]
```

Comment:
this formatting is a bit tragic to me.  Was ruff being too inflexible about line lengths?

Reply:
It looks like the line here goes up to 89 chars, I think ruff was being protective of this and choosing the latest spot to split on for some reason.

Reply:
ruff got rid of flake8's wiggle-room option for line lengths, which is a bit of a bummer

Reply:
I think for now I'll add an extra var assignment to get an assemblage that ruff doesn't massacre, in the future we can consider changing how flake8/ruff is configured here